### PR TITLE
Fixed condition for showing fields for Gutenberg only.

### LINKED ISSUE #2174: Course Information Settings Move
### Reproduction Steps

+ Courses -> Add New
+ The Course Length and Difficulty fields are visible under Course Options -> General
+ Save the course then view on the front end.
+ Jump back into edit the course, and now the Course Length field is now missing in Course Options > General and moved to the Course Information block where it belongs 

### Expected Behavior

+ It should always be under the Course Information block settings?

### Actual Behavior

+ It seems like the setting disappears from the user's view

### Error Messages / Logs

+ Include any relevant error messages or log files
```
<!-- Paste error logs / backtraces below this line -->

```

### System and Environment Information

<details>
<summary>System Report</summary>

```
Wordpress
-------------------------------------------

Home Url: https://dizzy-dust.myliftersite.com
Site Url: https://dizzy-dust.myliftersite.com
Login Url: https://dizzy-dust.myliftersite.com/wp-login.php
Version: 6.0
Debug Mode: No
Debug Log: No
Debug Display: Yes
Locale: en_US
Multisite: No
Page For Posts: Not Set
Page On Front: Not Set
Permalink Structure: /%postname%/
Show On Front: posts
Wp Cron: Yes


Settings
-------------------------------------------

Version: 6.6.0
Db Version: 
Course Catalog: Course Catalog (#13) [https://dizzy-dust.myliftersite.com/courses/]
Membership Catalog: Membership Catalog (#14) [https://dizzy-dust.myliftersite.com/memberships/]
Student Dashboard: Dashboard (#16) [https://dizzy-dust.myliftersite.com/dashboard/]
Checkout Page: Purchase (#15) [https://dizzy-dust.myliftersite.com/purchase/]
Course Catalog Per Page: 9
Course Catalog Sorting: menu_order
Membership Catalog Per Page: 9
Membership Catalog Sorting: menu_order
Site Membership: Not Set
Courses Endpoint: my-courses
Edit Endpoint: edit-account
Lost Password Endpoint: lost-password
Vouchers Endpoint: redeem-voucher
Autogenerate Username: no
Password Strength Meter: no
Minimum Password Strength: 
Terms Required: no
Terms Page: Not Set
Checkout Names: 
Checkout Address: 
Checkout Phone: 
Checkout Email Confirmation: no
Open Registration: no
Registration Names: 
Registration Address: 
Registration Phone: 
Registration Voucher: 
Registration Email Confirmation: no
Account Names: 
Account Address: 
Account Phone: 
Account Email Confirmation: no
Confirmation Endpoint: confirm-payment
Force Ssl Checkout: no
Country: US
Currency: USD
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Decimals: 2
Trim Zero Decimals: no
Recurring Payments: yes
Email From Address: team+sandbox@lifterlms.com
Email From Name: A LifterLMS Sandbox
Email Footer Text: 
Email Header Image: 
Cert Bg Width: 
Cert Bg Height: 
Cert Legacy Compat: 


Constants
-------------------------------------------

LLMS_REMOVE_ALL_DATA: undefined
LLMS_REST_DISABLE: undefined
LLMS_SITE_FEATURE_RECURRING_PAYMENTS: undefined
LLMS_SITE_IS_CLONE: undefined


Gateways
-------------------------------------------

Manual: Disabled
Manual Logging: 
Manual Order: 1


Server
-------------------------------------------

Mysql Version: 5.7.38
Php Curl: Yes
Php Default Timezone: UTC
Php Fsockopen: Yes
Php Max Input Vars: 5000
Php Max Upload Size: 512 MB
Php Memory Limit: 256M
Php Post Max Size: 1024M
Php Soap: Yes
Php Suhosin: No
Php Time Limt: 30
Php Version: 8.0.19
Software: Apache/2.4.53 (Unix) OpenSSL/1.1.1
Wp Memory Limit: 256M


Browser
-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.61 Safari/537.36


Theme
-------------------------------------------

Name: Twenty Twenty-Two
Version: 1.2
Themeuri: https://wordpress.org/themes/twentytwentytwo/
Authoruri: https://wordpress.org/
Template: 
Child Theme: No
Llms Support: No


Plugins
-------------------------------------------

LifterLMS: 6.6.0


Integrations
-------------------------------------------

BbPress: No
BuddyPress: No


Template Overrides
-------------------------------------------




```

</details>

This issue has be recreated:
+ [x] Locally
+ [ ] On a staging site
+ [x] On a production website
+ [x] With only LifterLMS and a default theme

### Browser, Device, and Operating System Information

+ Browser name and version
+ Operating System name and version
+ Device name and version (if applicable)

### PR DESCRIPTION
**Pull request by actuallyakash**
## Description
Hides two fields i.e., **Course Length** and **Difficulty** fields when the post is not migrated or the editor is Gutenberg. Those fields are visible only to the classic editor now.

Fixes #2174 

## How has this been tested?
Manually.

## Types of changes

Bug fix

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php [hunk 1]
```diff
@@ -356,7 +357,10 @@ public function get_fields() {
 			),
 		);
 
-		if ( function_exists( 'register_block_type' ) && llms_blocks_is_post_migrated( $this->post->ID ) ) {
+		if (
+			( ! class_exists( 'Classic_Editor' ) && 'auto-draft' === get_post_status( $this->post->ID ) ) ||
```
Comment (eri-trabiccolo) [inline includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php]: what if the classic editor exists (plugin installed and active) but the current user uses the block editor by default?
have you checked this?

Comment (actuallyakash) [reply includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php]: Thanks @eri-trabiccolo 

Didn't know about the switching aspect of the editors. Just pushed a fix, and now it's working fine.

File: includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php [hunk 2]
```diff
@@ -356,7 +357,18 @@ public function get_fields() {
...
+		$is_gutenberg   = method_exists( $current_screen, 'is_block_editor' ) && $current_screen->is_block_editor();
+
+		/**
+		 * Remove length and difficulty fields if
+		 * - the course is a new post and the editor is Gutenberg.
+		 * - the course is migrated to blocks used in the Gutenberg editor.
+		 */
+		if (
+			( 'auto-draft' === get_post_status( $this->post->ID ) && $is_gutenberg ) ||
+			$is_gutenberg && llms_blocks_is_post_migrated( $this->post->ID )
```
Comment (eri-trabiccolo) [inline includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php]: so, it's:
```suggestion
		$is_gutenberg &&
		( 
				'auto-draft' === get_post_status( $this->post->ID )  ||
				llms_blocks_is_post_migrated( $this->post->ID )
		)
```
?

File: includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php [hunk 3]
```diff
@@ -356,7 +357,18 @@ public function get_fields() {
 			),
 		);
 
-		if ( function_exists( 'register_block_type' ) && llms_blocks_is_post_migrated( $this->post->ID ) ) {
+		$current_screen = get_current_screen();
+		$is_gutenberg   = method_exists( $current_screen, 'is_block_editor' ) && $current_screen->is_block_editor();
```
Comment (eri-trabiccolo) [inline includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php]: This is a bit paranoid I know but what about
```suggestion
		$is_gutenberg   = is_object( $current_screen ) && method_exists( $current_screen, 'is_block_editor' ) && $current_screen->is_block_editor();
```

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
Minor changes requested, but looks good.
I'm gonna test it when ready <3

File: includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php (hunk 2 above)
Comment (actuallyakash) [reply includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php]: oh noice, thanks!

File: includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php (hunk 3 above)
Comment (actuallyakash) [reply includes/admin/post-types/meta-boxes/class.llms.meta.box.course.options.php]: you never know, prevention is always better than cure :D 

thanks, all done 👍

**Review by eri-trabiccolo — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**